src/ImageGallery.jsx

```diff
@@ -1016,6 +1018,10 @@ export default class ImageGallery extends React.Component {
         nextIndex = 0;
       }
 
+      if (onBeforeSlide) {
```

Comment:
There is an edge case on mobile swipe where the user does not swipe enough thus remaining on the same slide.
In this case you need to make sure that its actually sliding to the next index.
```
if (onBeforeSlide && nextIndex !== currentIndex) {
  onBeforeSlide(nextIndex);
}
```